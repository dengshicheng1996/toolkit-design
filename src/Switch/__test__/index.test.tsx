import * as React from 'react'
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Switch } from '..'

describe('Switch', () => {
  it('snapshot', () => {
    const component = mount(<Switch checked disabled data-y="111" />)
    expect(toJSON(component)).toMatchSnapshot()
  })

  it('normal', () => {
    const component = mount(<Switch checked disabled />)

    expect(component.find('input').prop('checked')).toBeTruthy()
    expect(component.find('input').prop('disabled')).toBeTruthy()
    expect(component.find('input').prop('value')).toBe(undefined)

    const component2 = mount(<Switch checked={false} value="a1" />)
    expect(component2.find('input').prop('checked')).toBeFalsy()
    expect(component2.find('input').prop('disabled')).toBeFalsy()
    expect(component2.find('input').prop('value')).toBe('a1')
  })
})
