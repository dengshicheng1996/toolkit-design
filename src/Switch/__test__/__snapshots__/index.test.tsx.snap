// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch snapshot 1`] = `
<Switch
  checked={true}
  data-y="111"
  defaultChecked={false}
  disabled={true}
>
  <div
    className="amd-switch"
    data-y="111"
  >
    <ForwardRef(C)
      checked={true}
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    >
      <ErrorBoundary
        onError={[Function]}
      >
        <Labelable
          checked={true}
          disabled={true}
          forwardRef={null}
          onChange={[Function]}
          type="checkbox"
        >
          <label>
            <input
              checked={true}
              disabled={true}
              onChange={[Function]}
              readOnly={true}
              style={
                Object {
                  "border": 0,
                  "display": "block",
                  "height": 0,
                  "margin": 0,
                  "overflow": "hidden",
                  "padding": 0,
                  "width": 0,
                }
              }
              type="checkbox"
            />
            <div
              className="amd-switch-fake amd-switch-fake-disabled amd-switch-fake-checked"
            />
          </label>
        </Labelable>
      </ErrorBoundary>
    </ForwardRef(C)>
  </div>
</Switch>
`;
